llvm/lib/Transforms/FuzzIntrospector/FuzzIntrospector.cpp

```diff
@@ -1132,3 +1184,234 @@ static RegisterStandardPasses
+
+        auto side0Comp = complexities.first;
+        auto side1Comp = complexities.second;
+
+        std::pair<std::string, std::string> dbgExtracts;
+        dbgExtracts = getInsnDebugInfo((Instruction *)BI);
+        auto BRstring = dbgExtracts.first;
+        if (BRstring.length() == 0) {
+          continue; // Failed to get debug info
+        }
+        dbgExtracts = getBBDebugInfo(side0, BILoc);
+        auto side0String = dbgExtracts.first;
+        if (side0String.length() == 0)
+          continue;
+        auto side0Line = std::stoi(dbgExtracts.second);
+        dbgExtracts = getBBDebugInfo(side1, BILoc);
+        auto side1String = dbgExtracts.first;
+        if (side1String.length() == 0)
+          continue;
+        auto side1Line = std::stoi(dbgExtracts.second);
+
+        // Decide on the sides based on line number distance; This is how coverage reports the sides
+        std::string trueSideString, falseSideString;
+        size_t trueSideCmp, falseSideComp;
+        if (side0Line > side1Line) {
+          trueSideString = side1String;
+          trueSideCmp = side1Comp;
+          falseSideString = side0String;
+          falseSideComp = side0Comp;
+        } else {
+          trueSideString = side0String;
+          trueSideCmp = side0Comp;
+          falseSideString = side1String;
+          falseSideComp = side1Comp;
+        }
+
+        BranchSidesComplexity entry_val(trueSideString, trueSideCmp, falseSideString, falseSideComp);
+        BranchProfileEntry entry = {BRstring, entry_val};
+        outMap.push_back(entry);
+
+      }
+    }
+
+  } // End of loop over M
+  writeOutMap(outMap, outFileName);
+}
+
+// Simple intra-procedural CFG traversal
+SmallPtrSet<BasicBlock *, 32> FuzzIntrospector::findReachables(BasicBlock *src) {
+  SmallVector<BasicBlock *, 32> worklist;
+  SmallPtrSet<BasicBlock *, 32> allReachables;
+
+  worklist.push_back(src);
+
+  while (!worklist.empty())
+  {
+    auto currBB = worklist.pop_back_val();
+
+    // This adds to the set and returns false if already was in the set: avoids loop
+    if (!allReachables.insert(currBB).second) {
+      continue;
+    }
+
+    if (auto TI = currBB->getTerminator()) {
+      for (unsigned i = 0, NSucc = TI->getNumSuccessors(); i < NSucc; ++i) {
+        worklist.push_back(TI->getSuccessor(i));
+      }
+    }
+  }
+
+  return allReachables;
+}
+
+// Calculate complexities reachable from each reachable unique BBs
+// TODO: for now, this function just accounts for complexity of callee functions from 
+// the reachable BBs. We can do better by carefully calculating complexity of the reachable
+// regions of the code i.e. intra-procedural complexity by counting the number of unique 
+// edges and nodes for each reachable set.
+std::pair<size_t, size_t> FuzzIntrospector::findComplexities(SmallPtrSet<BasicBlock *, 32> trueReachable, SmallPtrSet<BasicBlock *, 32> falseReachable, std::map<BasicBlock *, size_t> BBComplexityMap) {
```

Comment:
Please run `clang-format` on the code -- I think this line is a bit too long for `clang-formats` style

Reply:
Done, wow it made a lot of changes!

Reply:
Ah, I see -- quick question, did you try to rebuild it after the clang-format? I'm asking because sometimes the `#includes` will be re-arranged in a manner that breaks the build. It's quite annoying -- in either case I will rebuild, test and merge tomorrow but if you already know then let me know!

Reply:
Yes, I was curious about the many changes almost all over the cpp file, so tried to re-build the clang image and it finished successfully.